diff --git a/build/core/Makefile b/build/core/Makefile
index 9403f63..789815d 100644
--- a/build/core/Makefile
+++ b/build/core/Makefile
@@ -1320,7 +1320,9 @@ else
 endif
 
 ifneq ($(CM_BUILD),)
-    $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := true
+    # patched by Str1ker
+    #$(INTERNAL_OTA_PACKAGE_TARGET): backuptool := true
+    $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := false
 else
     $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := false
 endif
diff --git a/build/tools/releasetools/ota_from_target_files b/build/tools/releasetools/ota_from_target_files
index 93d1c23..58eeb5b 100755
--- a/build/tools/releasetools/ota_from_target_files
+++ b/build/tools/releasetools/ota_from_target_files
@@ -418,7 +418,8 @@ def WriteFullOTAPackage(input_zip, output_zip):
   if "selinux_fc" in OPTIONS.info_dict:
     WritePolicyConfig(OPTIONS.info_dict["selinux_fc"], output_zip)
 
-  script.FormatPartition("/system")
+  # patched by Str1ker
+  #script.FormatPartition("/system")
   script.Mount("/system")
   #script.UnpackPackageDir("recovery", "/system")
   script.UnpackPackageDir("system", "/system")
